packages/twenty-front/src/modules/action-menu/actions/record-actions/single-record/workflow-actions/hooks/useDeactivateWorkflowWorkflowSingleRecordAction.ts

```diff
@@ -0,0 +1,59 @@
+import { useActionMenuEntries } from '@/action-menu/hooks/useActionMenuEntries';
+import {
+  ActionMenuEntryScope,
+  ActionMenuEntryType,
+} from '@/action-menu/types/ActionMenuEntry';
+import { useDeactivateWorkflowVersion } from '@/workflow/hooks/useDeactivateWorkflowVersion';
+import { useWorkflowWithCurrentVersion } from '@/workflow/hooks/useWorkflowWithCurrentVersion';
+import { IconPlayerPause, isDefined } from 'twenty-ui';
+
+export const useDeactivateWorkflowWorkflowSingleRecordAction = ({
+  workflowId,
+}: {
+  workflowId: string;
+}) => {
+  const { addActionMenuEntry, removeActionMenuEntry } = useActionMenuEntries();
+
+  const { deactivateWorkflowVersion } = useDeactivateWorkflowVersion();
+
+  const workflowWithCurrentVersion = useWorkflowWithCurrentVersion(workflowId);
+
+  const isWorkflowActive =
+    isDefined(workflowWithCurrentVersion) &&
+    workflowWithCurrentVersion.currentVersion.status === 'ACTIVE';
+
+  const registerDeactivateWorkflowWorkflowSingleRecordAction = ({
+    position,
+  }: {
+    position: number;
+  }) => {
+    if (
+      !isDefined(workflowWithCurrentVersion) ||
+      !isDefined(workflowWithCurrentVersion.currentVersion.trigger) ||
```

Comment:
are you sure you need this check? A workflowVersion without trigger can not have status=ACTIVE